Add duck-duck-go-search extension

tegola opened the pull request:
## Description

Adds support for Duck Duck Go search with suggestions, with the option to save the search history locally.

Sorry for the messed up commit history, the `ray` command was continuosly asking me to pull contributions I had already merged, I had to recreate my own `extensions` repository fork to fix it.

## Screencast

<img width="1000" alt="duck-duck-go-search 2023-04-10 at 14 36 53" src="https://user-images.githubusercontent.com/1868590/230902673-08094702-5411-41f1-b0f1-0acc7e43ec00.png">

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/duck-duck-go-search/src/index.tsx:
```diff
@@ -0,0 +1,59 @@
+                  )}
+
+                  <Action
+                    title="Clear All History"
```
```suggestion
                    title="Clear All History"
                    style={Action.Style.Destructive}
```

pernielsentikaer commented on extensions/duck-duck-go-search/src/index.tsx:
```diff
@@ -0,0 +1,59 @@
+                <ActionPanel.Section title="History">
+                  {item.isHistory && (
+                    <Action
+                      title="Remove From History"
```
```suggestion
                      title="Remove from History"
                      style={Action.Style.Destructive}
```

pernielsentikaer commented on extensions/duck-duck-go-search/src/utils/useSearch.tsx:
```diff
@@ -0,0 +1,138 @@
+  }
+
+  async function deleteAllHistory() {
+    await LocalStorage.removeItem(HISTORY_KEY);
```
I would implement [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) here

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

* Seems like some throttle could be a good idea, it's bouncing a bit - not sure if it's possible to improve.

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

tegola commented:
Thanks for your suggestions, which I have implemented. I also changed the shortcuts and style/icons for deleting history items to better match with other plugins using an history (clipboard, calculator).

I didn't implement throttling because search seemed a little unresponsive.

justspacedog commented:
I had the same approch but with a little bit more functionality build in here: [https://github.com/raycast/extensions/pull/5754](https://github.com/raycast/extensions/pull/5754)

Maybe you can add this? I closed the pull request because I want to make sure there is no copyright on the icon, but it seems that everybody here doesn't care (and legal@duckduckgo.com doesn't answer me).

tegola commented:
I'm not an avid bangs user, and I feel they break the one-functionality-per-plugin approach in Raycast. Also, DuckDuckGo suggestions aren't even really related to the selected bang. But! If you think they're useful and like to contribute to my release, please do! And don't forget to add your name to the credits!

Regarding the copyright on the icon, I made a little research and ultimately just decided to use it thinking I'd be covered by a universal acceptable use policy.

justspacedog commented:
> I'm not an avid bangs user, and I feel they break the one-functionality-per-plugin approach in Raycast. Also, DuckDuckGo suggestions aren't even really related to the selected bang. But! If you think they're useful and like to contribute to my release, please do! And don't forget to add your name to the credits!

I use bangs all the time and think it's a good idea to actually see where you're searching. But you are right that duckduckgo's suggestions don't always match the suggestions on the other website.

> Regarding the copyright on the icon, I made a little research and ultimately just decided to use it thinking I'd be covered by a universal acceptable use policy.

I don't know. But I'd better ask, because I've heard a lot of strange things.

I hope I have done it right, here is the update: [https://github.com/raycast/extensions/pull/6062](https://github.com/raycast/extensions/pull/6062)

tegola commented:
> I hope I have done it right, here is the update: #6062

Your pull request actually replaced all files, but as long the unmodified files’ content is the same, I guess it’s ok?

I’m away from my computer, but I noticed it’s already been merged, so please make sure the published extension works as intended.

Thanks

tegola commented:
My bad, it hasn’t been merged. Cramped screen fooled me.

I’ll take a better look tomorrow, thanks.
